jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/DefaultCredentialRetrievers.java

```diff
@@ -90,15 +92,15 @@ public DefaultCredentialRetrievers setInferredCredential(
   }
 
   /**
-   * Sets the suffix for a known credential helper.
+   * Sets the known credential helper. May either be a path to a credential helper executable, or a
+   * credential helper suffix (following {@code docker-credential-}).
    *
-   * @param credentialHelperSuffix the known credential helper suffix (following {@code
-   *     docker-credential-})
+   * @param credentialHelper the path to a credential helper, or a credential helper suffix
+   *     (following {@code docker-credential-}) if the path doesn't exist.
    * @return this
    */
-  public DefaultCredentialRetrievers setCredentialHelperSuffix(
-      @Nullable String credentialHelperSuffix) {
-    this.credentialHelperSuffix = credentialHelperSuffix;
+  public DefaultCredentialRetrievers setCredentialHelper(@Nullable String credentialHelper) {
```

Comment:
I think we should have two methods, one for setting a suffix and one for specifying a path rather than overloading this single `credentialHelper` parameter.

Reply:
I think it's better to keep the logic for parsing user input (into the separate credential helper suffix vs. credential helper path) as close to the input as possible, rather than hiding the logic within this class. I think we will need to refactor more of the common plugin code later, so this can be here for now.